File: src/main/java/com/gandan/practice/factory/product/number/MobileNumber.java

package com.gandan.practice.factory.product.number;


public class MobileNumber extends Number {
    private String countryCode; 
    private String phoneNumber;

    public void setCountryCode(String countryCode){
        this.countryCode = countryCode;
    }
    
    public void setPhoneNumber(String phoneNumber){
        this.phoneNumber = phoneNumber;
    }
    
    @Override
    public long getNumber() {
        return Long.parseLong(this.countryCode+ this.phoneNumber);
    }
    
}

File: src/main/java/com/gandan/practice/factory/product/number/MobileNumber.kt
package com.gandan.practice.factory.product.number


class MobileNumber : Number() {
    var countryCode: String? = null
    var phoneNumber: String? = null

    override val number: Long
        get() = java.lang.Long.parseLong(this.countryCode!! + this.phoneNumber!!)

}
